src/pytorch_lightning/core/module.py

```diff
@@ -118,6 +122,10 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._register_sharded_tensor_state_dict_hooks_if_available()
         self._compiler_ctx: Optional[Dict[str, Any]] = None
 
+        # attributes only used when using fabric
+        self._fabric: Optional["lf.Fabric"] = None
+        self._optimizers: List[_FabricOptimizer] = []
```

Comment:
I would make it clear that this is fabric specific

```suggestion
        self._fabric_optimizers: List[_FabricOptimizer] = []
```